File: Backend/src/main/java/com/petadoption/model/RescueReport.java
package com.petadoption.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Document(collection = "rescue_reports")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RescueReport {

    @Id
    private String id;

    private String userEmail;
    private String reporterName;
    private String reporterPhone;
    private String animalType;
    private String description;
    private String location;
    private String address;
    private String urgency;
    private String imagePreview;
    private List<String> aiActions;
    private String aiSeverity;
    private Integer aiConfidence;
    private String rescueCenterId;
    private String rescueCenterName;

    private String status;   // Submitted, Acknowledged, In Progress, Rescued, Closed
    private String submittedAt;
    private String updatedAt;
    private List<RescueTimelineEvent> timeline;
}

File: Backend/models/RescueReport.js
const mongoose = require('mongoose');
const rescueReportSchema = new mongoose.Schema({
    userEmail: String,
    reporterName: String,
    reporterPhone: String,
    animalType: String,
    description: String,
    location: String,
    address: String,
    urgency: String,
    imagePreview: String,
    aiActions: [String],
    aiSeverity: String,
    aiConfidence: Number,
    rescueCenterId: String,
    rescueCenterName: String,
    status: { type: String, default: 'Pending' },
    submittedAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
    timeline: [{ event: String, date: Date, description: String }]
});
module.exports = mongoose.model('RescueReport', rescueReportSchema, 'rescue_reports');